### ISSUE TITLE
ios ModalUI显示相关问题

### ISSUE THREAD

**Issue by fishyumiao** [apache/incubator-weex#1493]
## Description

使用weex sdk 0.18.0发现 ‘modal.toast({ message: '123'})'显示不了
查看modalUIModule源码，发现model父view使用的不是keyWindow:
    UIView *superView =  [[[UIApplication sharedApplication] windows] objectAtIndex:0];
正好我的App有多个window，第一个window为hidden，故显示不了。

weex sdk更新到最新版本发现此处已经修改，并能正确显示’modal.toast‘窗口了。
    UIView *superView = self.weexInstance.rootView.window;
    if (!superView) {
        superView =  self.weexInstance.rootView;
    }

但是在 - (UIView *)toastViewForMessage:(NSString *)message superView:(UIView *)superView 中，
仍然又出现了：UIWindow *window = [[[UIApplication sharedApplication] windows] objectAtIndex:0];
既然有superView了，就可以用superView了，这里应该去掉！！！

另：
又往下面看了几行，发现switch中，每个分支都有：
    point = CGPointMake(window.frame.size.width/2, window.frame.size.height/2);
这也太不走心了，明显应该提出去的代码写进switch。。。。。。

还有weex sdk中 ’WXModuleCallback‘既然已经标识为deprecated，推荐使用’WXModuleKeepAliveCallback‘，为什么还不将所有WXModuleCallback替换掉？
搞得一堆warnning~~~用起来不爽啊~~~

## Environment

ios weex sdk 0.18.0

**Comment by fishyumiao** [comment]
WXListComponent.m文件中还包含以下代码：
```
- (void)fixFlicker
{
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        // FIXME:(ง •̀_•́)ง┻━┻ Stupid scoll view, always reset content offset to zero by calling _adjustContentOffsetIfNecessary after insert cells.
        // So if you pull down list while list is rendering, the list will be flickering.
        // Demo:    
        // Have to hook _adjustContentOffsetIfNecessary here.
        // Any other more elegant way?
        NSString *a = @"ntOffsetIfNe";
        NSString *b = @"adjustConte";
        
        NSString *originSelector = [NSString stringWithFormat:@"_%@%@cessary", b, a];
        [[self class] weex_swizzle:[WXTableView class] Method:NSSelectorFromString(originSelector) withMethod:@selector(fixedFlickerSelector)];
    });
}
```
a,b,originSelector三行代码雷到我了！！！为什么要这样写？

**Comment by xuyouyang** [comment]
@fishyumiao '_adjustContentOffsetIfNecessary' is a private method in UIScrollerView .I think you'd better learn more about this.

**Comment by xuyouyang** [comment]
@fishyumiao Yes, it is a easy and effective way to reduce risk of APP review.
